src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonNode/JsonArrayTests.cs

```diff
@@ -486,5 +487,210 @@ public static void LazyInitializationIsThreadSafe()
+            Assert.Equal("abcd", array[1].GetValue<string>());
+            Assert.Null(array[2]);
+            Assert.True(array[3].GetValue<bool>());
+            Assert.False(array[4].GetValue<bool>());
+            Assert.Equal(30, array[5].GetValue<int>());
+
+            JsonObject clonedNestedJObject = array[6].AsObject();
+            Assert.Equal(nestedJsonObj.Count, clonedNestedJObject.Count);
+            Assert.Equal(9, clonedNestedJObject["Nine"].GetValue<int>());
+            Assert.Equal("def", clonedNestedJObject["Name"].GetValue<string>());
+
+            JsonArray clonedNestedArray = clonedArray[7].AsArray();
+            Assert.Equal(nestedArray.Count, clonedNestedArray.Count);
+            Assert.Equal("elem0", clonedNestedArray[0].GetValue<string>());
+            Assert.Equal("elem1", clonedNestedArray[1].GetValue<string>());
+
+            string originalJson = array.ToJsonString();
+            string clonedJson = clonedArray.ToJsonString();
+
+            Assert.Equal(originalJson, clonedJson);
+        }
+
+        [Fact]
+        public static void DeepCloneFromElement()
+        {
+            using (JsonDocument document = JsonDocument.Parse("[\"abc\", 10]"))
+            {
+                JsonArray jArray = JsonArray.Create(document.RootElement);
+                var clone = jArray.DeepClone().AsArray();
+
+                Assert.True(JsonNode.DeepEquals(jArray, clone));
+                Assert.Equal(10, clone[1].GetValue<int>());
+                Assert.Equal("abc", clone[0].GetValue<string>());
+            }
+        }
+
+        [Fact]
+        public static void DeepEquals()
+        {
+            var array = new JsonArray() { null, 10, "str" };
+            var sameArray = new JsonArray() { null, 10, "str" };
+
+            Assert.True(JsonNode.DeepEquals(array, array));
+            Assert.True(JsonNode.DeepEquals(array, sameArray));
+            Assert.True(JsonNode.DeepEquals(sameArray, array));
+
+            Assert.False(JsonNode.DeepEquals(array, null));
+
+            var diffArray = new JsonArray() { null, 10, "s" };
+            Assert.False(JsonNode.DeepEquals(array, diffArray));
+            Assert.False(JsonNode.DeepEquals(diffArray, array));
+
+            diffArray = new JsonArray() { null, 10 };
+            Assert.False(JsonNode.DeepEquals(array, diffArray));
+            Assert.False(JsonNode.DeepEquals(diffArray, array));
+        }
+
+        [Fact]
+        public static void DeepEqualsWithJsonValueArrayType()
+        {
+            var array = new JsonArray();
+            array.Add(JsonValue.Create(2));
+            array.Add(JsonValue.Create(3));
+            array.Add(JsonValue.Create(4));
+            var value = JsonValue.Create(new long[] { 2, 3, 4 });
+
+            Assert.True(JsonNode.DeepEquals(array, value));
+        }
+
+        [Fact]
+        public static void DeepEqualsFromElement()
+        {
+            using (JsonDocument document = JsonDocument.Parse("[1, 2, 4]"))
```

Comment:
Nit: unless the test specifically cares about the document being disposed before the method returns I would recommend using modern using bindings:
```suggestion
            using JsonDocument document = JsonDocument.Parse("[1, 2, 4]");
```